db: correctly quote `index` columns to allow sqlite support

durin42 opened the pull request:
Yes, postgres is probably required for larger installations but I've
been successfully using the bridge for months with this patch applied,
so I figured I should send it upstream. As far as I know it shouldn't
hurt anything on other SQL engines, but I'm far from an expert.

durin42 committed:
db: correctly quote `index` columns to allow sqlite support

Yes, postgres is probably required for larger installations but I've
been successfully using the bridge for months with this patch applied,
so I figured I should send it upstream. As far as I know it shouldn't
hurt anything on other SQL engines, but I'm far from an expert.

sumnerevans commented on mautrix_googlechat/db/message.py:
```diff
@@ -49,7 +49,7 @@ def _from_row(cls, row: Record | None) -> Message | None:
     columns = (
         "mxid, mx_room, gcid, gc_chat, gc_receiver, gc_parent_id, "
-        "index, timestamp, msgtype, gc_sender"
+        "`index`, timestamp, msgtype, gc_sender"
```
```suggestion
        '"index", timestamp, msgtype, gc_sender'
```

It's better to use `"` as the delimiter within the SQL, and switch to `'` for the Python string delimiter.

According to https://stackoverflow.com/questions/34884362/escape-table-name-in-sqlite#34884445:

> `keyword` A keyword enclosed in grave accents (ASCII code 96) is an identifier. This is not standard SQL. This quoting mechanism is used by MySQL and is included in SQLite for compatibility.

(This applies to the entire PR)

durin42 replied:
Done throughout.

durin42 committed:
db: switch to using " instead of ` for quoting sql

This is evidently more standard, and ` is a thing sqlite supports
to be more compatible with MySQL.
